Repeat layer Dim needs to be defined

I am trying to test some repeat construction behavior and currently running into this issue: 

```
  File "/u/hilmes/experiments/tts_new_sis/recipe/returnn_common/nn/naming.py", line 876, in NetDictBuilderCtx.make_net_dict_raw
    line: assert dim.dyn_size_ext, f"need {dim} to be defined to be able to know about implicit dims"
    locals:
      dim = <local> Dim{'rep_time'[?]}
      dim.dyn_size_ext = <local> None
AssertionError: need Dim{'rep_time'[?]} to be defined to be able to know about implicit dims
```

with this test case: 
```python
def test_repeat_layer():
  nn.reset_default_root_name_ctx()
  feat = nn.FeatureDim('F', 3)
  time = nn.SpatialDim("time")
  rep_time = nn.SpatialDim("rep_time")
  data = nn.get_extern_data(nn.Data('data', dim_tags=[nn.batch_dim, time, feat]))
  const = nn.constant(value=5, shape=[nn.batch_dim, time])
  rep, _ = nn.repeat(data, repetitions=const, axis=time, out_dim=rep_time)
  rep.mark_as_default_output()
  config_code = nn.get_returnn_config().get_complete_py_code_str(nn.Module())
  from returnn_common.tests.returnn_helpers import dummy_run_net_single_custom
  dummy_run_net_single_custom(config_code)
```

The same error also occurs when not defining rep_time explicitly. I am most likely doing something wrong, because I was initially trying to replicate another error from a model (separate issue once I am able to replicate it in a small test), but after some trying I am not sure whats wrong with this example.

--- a/tests/test_nn_generated_layers.py
+++ b/tests/test_nn_generated_layers.py
@@ -38,3 +38,14 @@ def test_repeat_int():
   rep.mark_as_default_output()
   config = nn.get_returnn_config().get_complete_py_code_str(nn.Module())
   dummy_run_net_single_custom(config)
+
+
+def test_repeat_layer():
+  # https://github.com/rwth-i6/returnn_common/issues/163
+  nn.reset_default_root_name_ctx()
+  time_dim = nn.SpatialDim("time")
+  data = nn.get_extern_data(nn.Data('data', dim_tags=[nn.batch_dim, time_dim, nn.FeatureDim('F', 3)]))
+  const = nn.constant(value=5, shape=[nn.batch_dim, time_dim])
+  nn.repeat(data, repetitions=const, axis=time_dim)[0].mark_as_default_output()
+  config_str = nn.get_returnn_config().get_complete_py_code_str(nn.Module())
+  dummy_run_net_single_custom(config_str)

